pkg/kp/kv.go

```diff
@@ -294,6 +295,11 @@ func (c consulStore) WatchPods(keyPrefix string, quitChan <-chan struct{}, errCh
 					return
 				case errChan <- util.Errorf("Could not parse pod manifest at %s: %s. Content follows: \n%s", pair.Key, err, pair.Value):
 				}
+				// do not send the partially built list, since we know it's incomplete
+				// and some callers may care about elements being removed
+				// TODO(sjung): those callers should probably be refactored to
+				// use NewKeyWatcher instead
+				continue WATCH_LOOP
```

Comment:
> they will investigate when it goes wrong

In general, manual intervention doesn't scale, and it makes the system more difficult to operate. We should never fall back to "a human will fix it" unless we truly can't avoid it. Humans can spend many hours failing to notice an "obvious" problem.

> a person is not going to pay attention

Is my deploy taking a long time, or is the process blocked?  I don't think the users of this system will necessarily be able to distinguish the two without resorting to a timeout, even if they are paying attention.  And until this hypothetical user asks for manual intervention, the host is effectively down for _all_ users.  Failure isolation is a very useful property to have.

> the alternative is [do nothing].

We can brainstorm for real alternatives offline.

> outage for the service being deployed

A service outage is not necessarily the outcome from this bug being triggered, especially because our standard deployment tools rely on a node coming up healthy before moving on.

> The long term solution is [...], but that's going to take more time to get right.

It's worth the effort.